Priority Queue

```diff
@@ -0,0 +1,38 @@
+class PriorityQueue(object): 
```

Comment:
you are pushing all commits to the same branch which makes having multiple PR redundant.